libraries/AP_BLHeli/AP_BLHeli.cpp

```diff
@@ -1296,6 +1296,26 @@ bool AP_BLHeli::get_telem_data(uint8_t esc_index, struct telem_data &td)
     return true;
 }
 
+// return the average motor frequency in Hz for dynamic filtering
+float AP_BLHeli::get_average_motor_frequency_hz() const
+{
+    float motor_freq = 0.0f;
+    uint32_t now = AP_HAL::millis();
+    uint8_t valid_escs = 0;
+    // average the rpm of each motor as reported by BLHeli and convert to Hz
+    for (uint8_t i = 0; i < num_motors; i++) {
+        if (last_telem[i].timestamp_ms && (now - last_telem[i].timestamp_ms < 1000)) {
+            valid_escs++;
+            motor_freq += last_telem[i].rpm / 60.0f;
```

Comment:
You could do:
`motor_freq += last_telem[i].rpm * 0.01666666666666; // use multiplication instead of division by 60 to convert rpm into Hz`

Reply:
While I can see that this may be slightly faster I do not like replacing exact maths with approximations unless there is an extremely compelling reason.